pkg/occlient/occlient.go

```diff
@@ -1290,22 +1303,26 @@ func (c *Client) PatchCurrentDC(name string, dc appsv1.DeploymentConfig, prePatc
 	}
 
 	// Replace the current spec with the new one
-	currentDC.Spec = dc.Spec
+	modifiedDC.Spec = dc.Spec
 
 	// Replace the old annotations with the new ones too
 	// the reason we do this is because Kubernetes handles metadata such as resourceVersion
 	// that should not be overridden.
-	currentDC.ObjectMeta.Annotations = dc.ObjectMeta.Annotations
-	currentDC.ObjectMeta.Labels = dc.ObjectMeta.Labels
+	modifiedDC.ObjectMeta.Annotations = dc.ObjectMeta.Annotations
+	modifiedDC.ObjectMeta.Labels = dc.ObjectMeta.Labels
 
 	// Update the current one that's deployed with the new Spec.
 	// despite the "patch" function name, we use update since `.Patch` requires
 	// use to define each and every object we must change. Updating makes it easier.
-	_, err := c.appsClient.DeploymentConfigs(c.Namespace).Update(currentDC)
+	updatedDc, err := c.appsClient.DeploymentConfigs(c.Namespace).Update(&modifiedDC)
 	if err != nil {
 		return errors.Wrapf(err, "unable to update DeploymentConfig %s", name)
 	}
 
+	if reflect.DeepEqual(updatedDc.Spec.Template, currentDC.Spec.Template) {
```

Comment:
It might be worth to add a comment explaining what lead to this and why are we doing this. I'm sure that we will forget it after some time ;-)